live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/properties/Storage.java

```diff
@@ -28,8 +28,34 @@
 import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.NotNull;
 
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.springframework.boot.context.properties.NestedConfigurationProperty;
+
 public record Storage(@NotNull Location location)
 {
-    public record Location(@NotBlank String endpoint)
-    {}
+    @Data
+    @Getter(AccessLevel.NONE)
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class Location
```

Comment:
Can this be a record too?

Reply:
@krdabrowski I came up with something like this (seems to work):
```
public record Storage(@NotNull Location location)
{
    public record Location (
            @NotBlank
            String endpoint,
            @NotNull
            Retry retry

    )
    {
        public Location
        {
            retry = Objects.requireNonNullElseGet(retry, Retry::new);
        }
    }
}
```
wdyt?

Reply:
Turned out it can - fixed.
@Pawel-608 thanks for hint!